src/modules/attributes.ts

```diff
@@ -26,28 +26,31 @@ function updateAttrs(oldVnode: VNode, vnode: VNode): void {
   if (oldAttrs === attrs) return;
   oldAttrs = oldAttrs || {};
   attrs = attrs || {};
+  const lowAttrs: any = {};
 
   // update modified attributes, add new attributes
   for (key in attrs) {
     cur = attrs[key];
-    old = oldAttrs[key];
+    const lowkey = key.toLowerCase();
```

Comment:
Agreed. We can always add the most common attributes to the typings in lowercase form too, to limit errors (and people usually know that custom data attributes are in lower case). Personally, I will never use server-side rendering but with this change I would still have to pay the price of lowercasing every single attribute of my app, on every render (twice).